File: src/main/java/com/matag/admin/game/game/Game.java
package com.matag.admin.game.game;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

import com.matag.admin.game.session.GameSession;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@ToString(exclude = "gameSessions")
@EqualsAndHashCode(exclude = "gameSessions")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "game")
public class Game {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private LocalDateTime createdAt;
  @Enumerated(EnumType.STRING)
  private GameType type;
  @Enumerated(EnumType.STRING)
  private GameStatusType status;
  @Enumerated(EnumType.STRING)
  private GameResultType result;
  private LocalDateTime finishedAt;

  @OneToMany(mappedBy = "game", cascade = CascadeType.REMOVE)
  private List<GameSession> gameSessions;
}

File: src/main/kotlin/com/matag/admin/game/game/Game.kt
package com.matag.admin.game.game

import com.matag.admin.game.session.GameSession
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "game")
data class Game(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Enumerated(EnumType.STRING)
    var type: GameType? = null,

    @Enumerated(EnumType.STRING)
    var status: GameStatus? = null,

    @Enumerated(EnumType.STRING)
    var result: GameResultType? = null,

    var finishedAt: LocalDateTime? = null,

    @OneToMany(mappedBy = "game", cascade = [CascadeType.REMOVE], orphanRemoval = true)
    var gameSessions: MutableList<GameSession> = mutableListOf()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Game) return false
        return id != null && id == other.id
    }

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String = "Game(id=$id, type=$type, status=$status)"
}